Adding documentation for Topology Aware Hints (#27032)
kubernetes/website: content/en/docs/tasks/administer-cluster/enabling-topology-aware-hints.md

---
reviewers:
- robscott
title: Enabling Topology Aware Hints

> content_type: task
> ---

Reviewer: ```suggestion
content_type: task
min-kubernetes-server-version: 1.21
---
```

Author: Updated, thanks!

<!-- overview -->
{{< feature-state for_k8s_version="v1.21" state="alpha" >}}

_Topology Aware Hints_ enable topology aware routing with topology hints
included in EndpointSlices. This approach tries to keep traffic close to where
it originated from. This can result in decreased costs and improved performance.

## {{% heading "prerequisites" %}}

>   {{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}

Reviewer: As you're using the version check shortcode you need to set `min-kubernetes-server-version` in the front matter.

Author: Updated, thanks!

The following prerequisites are needed in order to enable topology aware hints:

>    * Kubernetes 1.21 or later

Reviewer: The version check shortcode covers this for you if you set `min-kubernetes-server-version` in the front matter.

Author: Updated, thanks!

   * {{< glossary_tooltip text="Kube-proxy" term_id="kube-proxy" >}} running in
     iptables mode or IPVS mode

>    * [Enable Endpoint
>      Slices](/docs/tasks/administer-cluster/enabling-endpointslices)

Reviewer: nit; I'd write:
```suggestion
   * Make sure that you have [enabled EndpointSlices](/docs/tasks/administer-cluster/enabling-endpointslices/)
```

Author: This is a bit of a strange thing to communicate since they're already on by default. I've switched this up to read "Ensure that you have not disabled EndpointSlices", does that make sense?

Reviewer: We could actually omit `/docs/tasks/administer-cluster/enabling-endpointslices/` from v1.21 I think: I don't think many cluster operators are turning this API off.

Reviewer: Also, this isn't as key a feature as EndpointSlice so if you want to, it's OK to omit `content/en/docs/tasks/administer-cluster/enabling-topology-aware-hints.md` from the PR entirely.

Author: Good call, removed this reference entirely. I'm not sure what the bar is for a page like this, but I think it could be a feature that users are interested in testing out early. Happy to remove it if that's preferred.

## Enable Topology Aware Hints

> To enable service topology, enable the `TopologyAwareHints` feature gate for

Reviewer: ```suggestion
To enable service topology, enable the `TopologyAwareHints` [feature gate](docs/reference/command-line-tools-reference/feature-gates/) for the
```

Author: Updated, thanks!

kube-apiserver, kube-controller-manager, and kube-proxy:

```
--feature-gates="TopologyAwareHints=true"
```

## {{% heading "whatsnext" %}}

> * Read about [Topology Aware Hints](/docs/concepts/services-networking/topology-aware-hints)
> * Read about [Endpoint Slices](/docs/concepts/services-networking/endpoint-slices)

Reviewer: nit:
```suggestion
* Read about [Topology Aware Hints](/docs/concepts/services-networking/topology-aware-hints/) for Services
* Read about [EndpointSlices](/docs/concepts/services-networking/endpoint-slices/)
* ```

Author: Updated, thanks!
